src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -716,25 +702,44 @@ bool emberAfOperationalCredentialsClusterAddNOCCallback(app::CommandHandler * co
     // Creating the initial ACL must occur after the PASE session has adopted the fabric index
     // (see above) so that the concomitant event, which is fabric scoped, is properly handled.
     err = CreateAccessControlEntryForNewFabricAdministrator(commandObj->GetSubjectDescriptor(), fabricIndex,
-                                                            commandData.caseAdminNode);
+                                                            commandData.caseAdminSubject);
     VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
 
+    // The Fabric Index associated with the armed fail-safe context SHALL be updated to match the Fabric
+    // Index just allocated.
+    err = failSafeContext.SetAddNocCommandInvoked(fabricIndex);
```

Comment:
Moving this here is pretty worrisome.  Now we can end up in a bad state where the new NOC is in the fabric table but we have not flagged the fact that AddNOC happened on the fail-safe context, if something between there and here errors out.  This is why this code used to be right after the `AddNewFabric` call.

Maybe what we should do is roll back the fabric table addition on failure of anything between that and this point?

Reply:
This is not worrisome because we have not commited fully and the fail-safe will expire. What would be worse is an error occurring where we mark AddNOC done, but it's not actually done and you have to await fail-safe expiry before trying again, because you could not provide a good value.

Overall, the root cause fix is to make sure there can never be partial uncommitted FabricTable state that could be used post reset

Reply:
Left this as is.

Reply:
Let's do a follow-up issue

Reply:
Will be fixed as part of https://github.com/project-chip/connectedhomeip/issues/15585